### ISSUE TITLE
hzn unregister on macOS displays error message

### ISSUE THREAD

**Issue by bmpotter** [open-horizon/anax#1908]
When i run `hzn unregister -f` on my mac, i **sometimes** get this error msg:

```
Unregistering this node, cancelling all agreements, stopping all workloads, and restarting Horizon...
Waiting for Horizon node unregister to complete: No Timeout specified ...
Can't connect to the Horizon REST API to run DELETE http://localhost:8081/node?block=true. Run 'systemctl status horizon' to check if the Horizon agent is running. Or set HORIZON_URL to connect to another local port that is connected to a remote Horizon agent via a ssh tunnel. Specific error is: Delete "http://localhost:8081/node?block=true": EOF
The node was not successfully unregistered, please use 'hzn unregister -D' to ensure the node is completely reset.
```

The unregister still seems to complete, because when i run `hzn node list` after this error msg, it is `unconfigured` .

**Comment by Anukriti2512** [comment]
I would like to take up this issue. Could you please assign this to me @linggao @dabooz ? Thank you

**Comment by dabooz** [comment]
After a ton of investigation into the problem, and being unable to recreate the problem at will (even though this problem has been seen enough times that we know it is a real problem) the theory is that the problem is temporal within the golang connection management layer and thus we are going to set request.Close = true (on the HTTP request object). This is full well knowing that the need for this setting was supposedly fixed in the golang compiler. The full agent has always had this setting and does not experience problems with similar symptoms. With this fix, the CLI will have the same setting.

**Comment by bmpotter** [comment]
There is a 2nd part to this issue: note that when a problem *does* occur trying to reach the agent, part of the error msg suggests running `systemctl status horizon`, but that doesn't exist on macOS.

**Comment by dabooz** [comment]
Issue #1611 will address the error message on macos.

**Comment by diru1100** [comment]
Is this issue solved? I think pr #2186 closed issue #1611 
If not, I would like to work on this.

**Comment by linggao** [comment]
@diru1100 yes, it is resolved.  When looking for issues to work on, please check the "Pipelines" for the issue. If the it is "In Progess" or "Review/QA", that means it is being worked by someone else.
